src/Chapter3.hs

```diff
@@ -1006,6 +1131,107 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+{- ORMOLU_ENABLE -}
+newtype StatHealth = StatHealth Int deriving (Show)
+
+newtype StatAttack = StatAttack Int deriving (Show)
+
+newtype StatDefense = StatDefense Int deriving (Show)
+
+class Fighter a where
+  actions :: (Fighter b) => a -> [b -> (Maybe a, Maybe b)]
```

Comment:
You use `(Maybe a, Maybe b)` a lot. You can move it into a separate data type and also write a few helper functions for that 🙂